src/main/webapp/WEB-INF/view/chat.jsp

```diff
@@ -58,7 +58,7 @@ List<Message> messages = (List<Message>) request.getAttribute("messages");
               //Iterate through responseJson with new messages
               $.each(responseJson, function(index, message) {
                   //Append them
-                  $($ul).append("<li><strong>" + message.author + "</strong>" + ': ' + message.content + "</li>");
+                  $($ul).append("<li><strong><a href=/users/" + message.author + " >" + message.author + "</a>" + ': ' + "</strong>" + message.content + "<small><sub> " + message.creation + "</sub></small>" + "</li>");
```

Comment:
Concatenation of more than ~2 Strings with the "+" operator is another of my favourite "code smells". Understand this creates many temporary Strings left for the garbage collector to clean up; in fact it is O(n**2)!! Definitely don't do this in an interview, always user StringBuilder !

Reply:
So after making some investigation about `string concatenation in javascript` I found that it was more efficient to build the string using this operator `+=` and building it before using it in the `DOM`. So I created a variable called `result` and built the string there, after that I appended the variable result in the list.

Reply:
Oh silly me, of course this is in javascript not java!